ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/fragment/AEMArticleItemFragment.java

```diff
@@ -0,0 +1,177 @@
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        Bundle args = getArguments();
+        if (args != null) {
+            mArticleKey = Uri.parse(args.getString(ARTICLE_KEY_TAG));
+        }
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_aem_article_item, container, false);
+        ButterKnife.bind(this, view);
+        return view;
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        mAemWebView.setWebViewClient(mWebViewClient);
+
+        mHandler.post(mWebViewRunnable);
+    }
+
+    //endregion LifeCycle
+
+    //region Article Data
+
+    private Article getArticleFromKey() {
+        if (mRDb == null) {
+            mRDb = ArticleRoomDatabase.getInstance(getContext().getApplicationContext());
+        }
+        return mRDb.articleDao().find(mArticleKey);
+    }
+
+    public void setArticle(Article article) {
+        this.mArticle = article;
+
+        setFragmentViews();
+    }
+
+    private void setFragmentViews() {
+        if (mArticle.content != null) {
+            loadWebViewData();
+        }
+        Objects.requireNonNull(getActivity()).setTitle(mArticle.title);
+    }
+
+    private void loadWebViewData() {
+        StringBuilder builder = new StringBuilder(mArticle.content);
+        builder.insert(mArticle.content.indexOf("<head>") + 7,
+                "<style> img { max-width: 100%; } </style>");
+        mAemWebView.loadDataWithBaseURL(mArticleKey.getHost(), builder.toString(),
+                "text/html", null, null);
+    }
+
+    private void getArticleAttachments() {
+        AttachmentRepository repository = new AttachmentRepository(getContext().getApplicationContext());
+        mArticleAttachments = repository.getAttachmentsByArticle(mArticleKey.toString()).getValue();
+    }
+
+    //endregion Article Data
+
+    //region WebClient
+
+    private AEMWebViewClient mWebViewClient = new AEMWebViewClient();
+
+    private class AEMWebViewClient extends WebViewClient {
+        @Nullable
+        @Override
+        public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
+            if (url.endsWith(".jpg")) {
+                //TODO: Load local Image WebResource
+            } else if (url.endsWith(".css")) {
+                //TODO: Load CSS WebResource
+            }
+
+            return super.shouldInterceptRequest(view, url);
+        }
+    }
+    //endregion WebClient
+
+    //region Runnable and Handler
+
+    private Runnable mWebViewRunnable = new Runnable() {
+        @Override
+        public void run() {
+            AsyncTask.execute(() -> {
+                mArticle = getArticleFromKey();
```

Comment:
Is there a better way to run on the background thread than AsyncTask.execute()

Reply:
actually, name it `findAsync` since it will asynchronously find the article

Reply:
I changed it I just haven't updated that commit yet